include/vcpkg/dependencies.h

```diff
@@ -144,14 +146,18 @@ namespace vcpkg
 
     struct CreateInstallPlanOptions
     {
-        CreateInstallPlanOptions(GraphRandomizer* r, Triplet t) : randomizer(r), host_triplet(t) { }
-        CreateInstallPlanOptions(Triplet t, UnsupportedPortAction action = UnsupportedPortAction::Error)
-            : host_triplet(t), unsupported_port_action(action)
+        CreateInstallPlanOptions(GraphRandomizer* r, Triplet t, const Path& p)
+            : randomizer(r), host_triplet(t), packages_dir(p)
+        {
+        }
+        CreateInstallPlanOptions(Triplet t, const Path& p, UnsupportedPortAction action = UnsupportedPortAction::Error)
+            : host_triplet(t), packages_dir(p), unsupported_port_action(action)
         {
         }
 
         GraphRandomizer* randomizer = nullptr;
         Triplet host_triplet;
+        Path packages_dir;
         UnsupportedPortAction unsupported_port_action = UnsupportedPortAction::Warn;
```

Comment:
The NSDMIs make this confusing. Is the default Error or Warning?

OK if you say this is existing and don't want to change it.

```suggestion
        CreateInstallPlanOptions(GraphRandomizer* r, Triplet t, const Path& p)
            : randomizer(r)
            , host_triplet(t)
            , packages_dir(p)
            , unsupported_port_action(UnsupportedPortAction::Warn)
        {
        }
        CreateInstallPlanOptions(Triplet t, const Path& p, UnsupportedPortAction action = UnsupportedPortAction::Error)
            : randomizer(nullptr)
            , host_triplet(t)
            , packages_dir(p)
            , unsupported_port_action(action)
        {
        }

        GraphRandomizer* randomizer;
        Triplet host_triplet;
        Path packages_dir;
        UnsupportedPortAction unsupported_port_action;
```